solr/core/src/test/org/apache/solr/core/TestFileSystemConfigSetService.java

```diff
@@ -73,8 +73,14 @@ public void testIgnoresFileUploadsOutsideOfConfigSetDirectory() throws IOExcepti
     // surface validation errors to enable bulk uploading
     final var invalidFilePaths =
         List.of(
-            ".." + File.separator + "escapePath",
-            "foo" + File.separator + ".." + File.separator + ".." + File.separator + "bar");
+            ".." + FileSystems.getDefault().getSeparator() + "escapePath",
+            "foo"
+                + FileSystems.getDefault().getSeparator()
+                + ".."
+                + FileSystems.getDefault().getSeparator()
+                + ".."
+                + FileSystems.getDefault().getSeparator()
+                + "bar");
     for (String invalidFilePath : invalidFilePaths) {
       fileSystemConfigSetService.uploadFileToConfig(configName, invalidFilePath, testdata, true);
```

Comment:
The javadocs of `uploadFileToConfig` (which I recall writing), specify that the path string provided is forward slash since it's in the abstract namespace of the config, not on the file system (necessarily).